egs/wsj/s5/steps/data/reverberate_data_dir.py

```diff
@@ -0,0 +1,352 @@
+
+
+def ParseFileToDict(file, assert2fields = False, value_processor = None):
+    if value_processor is None:
+        value_processor = lambda x: x[0]
+
+    dict = {}
+    for line in open(file, 'r'):
+        parts = line.split()
+        if assert2fields:
+            assert(len(parts) == 2)
+
+        dict[parts[0]] = value_processor(parts[1:])
+    return dict
+
+
+# This is the major function to generate pipeline command for the corruption
+# The generic command of wav-reverberate will be like:
+# wav-reverberate --duration=t --impulse-response=rir.wav 
+# --additive-signals='noise1.wav,noise2.wav' --snrs='snr1,snr2' --start-times='s1,s2' input.wav output.wav
+def CorruptWav(wav_scp, durations, output_dir, room_dict, noise_list, foreground_snr_array, background_snr_array, num_replicas, prefix, speech_rvb_probability, noise_adding_probability, max_noises_per_minute):
+    foreground_snrs = list_cyclic_iterator(foreground_snr_array)
+    background_snrs = list_cyclic_iterator(background_snr_array)
+    command_list = []
+    for i in range(num_replicas):
+        keys = wav_scp.keys()
+        keys.sort()
+        for wav_id in keys:
+            wav_pipe = wav_scp[wav_id]
+            # check if it is really a pipe
+            if len(wav_pipe.split()) == 1:
+                wav_pipe = "cat {0} |".format(wav_pipe)
+            speech_dur = durations[wav_id]
+            max_noises_recording = math.floor(max_noises_per_minute * speech_dur / 60)
+            if prefix is not None:
+                wav_id = prefix + str(i) + "_" + wav_id
+
+            # pick the room
+            room = PickItemWithProbability(room_dict)
+            command_opts = ""
+            noises_added = []
+            snrs_added = []
+            start_times_added = []
+            if random.random() < speech_rvb_probability:
+                # pick the RIR to reverberate the speech
+                speech_rir = PickItemWithProbability(room.rir_list)
+                command_opts += "--impulse-response={0} ".format(speech_rir.rir_file_location)
+                # add the corresponding isotropic noise if there is any
+                if len(speech_rir.iso_noise_list) > 0:
+                    isotropic_noise = PickItemWithProbability(speech_rir.iso_noise_list)
+                    # extend the isotropic noise to the length of the speech waveform
+                    noises_added.append("wav-reverberate --duration={1} {0} - |".format(isotropic_noise.noise_file_location, speech_dur))
+                    snrs_added.append(background_snrs.next())
+                    start_times_added.append(0)
+
+            # Add the point-source noise
+            if len(noise_list) > 0 and random.random() < noise_adding_probability:
+                for k in range(random.randint(1, max_noises_recording)):
+                    # pick the RIR to reverberate the point-source noise
+                    noise = PickItemWithProbability(noise_list)
```

Comment:
I am assuming that this scripts allows for simple addition of background noises without reverberation. Is this correct ?